### PR TITLE
WIP - single node serial workflow

### PR DESCRIPTION
**Pull request by omertuc**
(Previously was called `WIP - single node serial and upgrade workflows` but upgrade flow was moved to [this PR](https://github.com/openshift/release/pull/17519))

Please ignore this for now.

(This change isn't really currently specific to the MCO, I just chose MCO arbitrarily because it had an existing openshift-e2e-aws-serial job I wanted to copy)

### CODE REVIEW

File: ci-operator/step-registry/openshift/upgrade/aws/single-node/openshift-upgrade-aws-single-node-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    - ref: openshift-e2e-test
+    post:
+    - ref: openshift-e2e-test-capabilities-check
+    - chain: gather-network
+    - chain: gather-core-dump
+    - chain: ipi-aws-post
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: "release:initial"
+      OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE: "release:latest"
+    env:
```
Comment (vrutkovs) [inline ci-operator/step-registry/openshift/upgrade/aws/single-node/openshift-upgrade-aws-single-node-workflow.yaml:20]: We should add `TEST_SKIPS` here - most of the disruption tests are invalid for SNO

**Comment by vrutkovs** [comment]
Reference: https://issues.redhat.com/browse/OTA-379

File: ci-operator/step-registry/openshift/upgrade/aws/single-node/openshift-upgrade-aws-single-node-workflow.yaml (hunk 1 above)
Comment (omertuc) [reply ci-operator/step-registry/openshift/upgrade/aws/single-node/openshift-upgrade-aws-single-node-workflow.yaml:20]: I see you already added it in yours. Updating this PR to only focus on serial

**Comment by ingvagabund** [comment]
@omertuc can you please rebase the PR to get the latest failing tests?

**Comment by ingvagabund** [comment]
```
[BeforeEach] PodTopologySpread Filtering
  k8s.io/kubernetes@v1.20.0/test/e2e/scheduling/predicates.go:788
STEP: Trying to get 2 available nodes which can run pod
```

`ci/rehearse/openshift/machine-config-operator/master/e2e-aws-single-node-serial` is running tests from [openshift-4.7 branch](https://github.com/openshift/kubernetes/blob/release-4.7/test/e2e/scheduling/predicates.go#L788), not the master branch.

**Comment by omertuc** [comment]
> ```
> [BeforeEach] PodTopologySpread Filtering
>   k8s.io/kubernetes@v1.20.0/test/e2e/scheduling/predicates.go:788
> STEP: Trying to get 2 available nodes which can run pod
> ```
> 
> `ci/rehearse/openshift/machine-config-operator/master/e2e-aws-single-node-serial` is running tests from [openshift-4.7 branch](https://github.com/openshift/kubernetes/blob/release-4.7/test/e2e/scheduling/predicates.go?rgh-link-date=2021-04-19T08%3A11%3A55Z#L788), not the master branch.
> 
> The `BeforeEach` in the master branch is expected to be reported [at line 720](https://github.com/openshift/kubernetes/blob/master/test/e2e/scheduling/predicates.go?rgh-link-date=2021-04-19T08%3A11%3A55Z#L720).

Odd, I'll look into it. Nice catch

**Comment by ingvagabund** [comment]
https://github.com/openshift/origin/pull/26054 needs to merge first

**Comment by ingvagabund** [comment]
@omertuc https://bugzilla.redhat.com/show_bug.cgi?id=1934085#c8 for the sig-scheduling tests. Some are still failing but the underlying reason is different. `image-registry` needs to create only a single instance of the pod instead of two.

**Comment by ingvagabund** [comment]
@omertuc can you rebase the PR? To see the latest failing tests (if any)?

**Comment by omertuc** [comment]
> @omertuc can you rebase the PR? To see the latest failing tests (if any)?

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/16290/rehearse-16290-pull-ci-openshift-machine-config-operator-master-e2e-aws-single-node-serial/1410245257366343680

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml [hunk 2]
```diff
@@ -461,6 +461,71 @@ presubmits:
...
+              name: cluster-secrets-aws
+      - name: pull-secret
+        secret:
+          secretName: registry-pull-credentials
+      - name: result-aggregator
+        secret:
+          secretName: result-aggregator
+    trigger: (?m)^/test( | .* )e2e-aws-single-node-serial,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (kikisdeliveryservice) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:525]: why did you change the contents of the next test to always_run:true?

can you clean this up so that you add your stuff with a completely new entry beginning with a new -agent

**Review by kikisdeliveryservice — CHANGES_REQUESTED** [review]

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml (hunk 2 above)
Comment (omertuc) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: Just bad rebase, didn't mean to. Should be better now

**Comment by kikisdeliveryservice** [comment]
looks good, @omertuc 

a little confused, do you want this e2e in this repo? initial message seems to say it's a test. ping me if/when you want approval. :smile:

**Comment by omertuc** [comment]
> looks good, @omertuc
> 
> a little confused, do you want this e2e in this repo? initial message seems to say it's a test. ping me if/when you want approval & remove WIP. 😄

It is just a WIP test, just didn't want to leave the ci-operator/jobs file messed up

**Comment by romfreiman** [comment]
@omertuc so lets add the serial job to the test grid of sno 4.9?

**Comment by omertuc** [comment]
> @omertuc so lets add the serial job to the test grid of sno 4.9?

@romfreiman 
https://github.com/kubernetes/test-infra/pull/22757
https://github.com/openshift/release/pull/19878

**Comment by omertuc** [comment]
Obsoleted by #19878 , closing
